script.js

```diff
@@ -1,12 +1,68 @@
 //You can edit ALL of the code here
 function setup() {
-  const allEpisodes = getAllEpisodes();
   makePageForEpisodes(allEpisodes);
 }
+const rootElem = document.getElementById("root");
+const allEpisodes = getAllEpisodes();
 
 function makePageForEpisodes(episodeList) {
-  const rootElem = document.getElementById("root");
-  rootElem.textContent = `Got ${episodeList.length} episode(s)`;
+  rootElem.innerHTML = "";
+  const display = document.getElementById("display");
+  display.innerText = `Display ${episodeList.length}/${allEpisodes.length}`;
+  episodeList.forEach(e => {
+  const boxBody = document.createElement('div');
+  boxBody.id = "boxBody";
+  const title = document.createElement('h3');
+  title.id = "title";
+  const imgBox = document.createElement('img');
+  imgBox.id = "imgBox";
+  const para = document.createElement('p');
+  para.id = "para"
+  rootElem.appendChild(boxBody);
+  boxBody.appendChild(title);
+  boxBody.appendChild(imgBox);
+  boxBody.appendChild(para);
+  title.innerHTML = `${e.name}`
+  +`-S${e.season.toString().padStart(2,'0')}E${e.number.toString().padStart(2,'0')}`;
+  imgBox.src = e.image.medium;
+  para.innerHTML = e.summary;
+});
 }
 
+//setting the filterd search results 
+let searchBar = document.getElementById("search");
+searchBar.addEventListener("keyup", function(p){
+  let filteredValue = p.target.value;
+ let filteredList = allEpisodes.filter((result)=>{
+   let name = result.name;
+   let text = result.summary;
+  return name.toLowerCase().includes(filteredValue)||text.toLowerCase().includes(filteredValue)
+  })
+  makePageForEpisodes(filteredList);
+});
+ //display the Episodes on the selector menu
+ const myList = document.getElementById("list");
+allEpisodes.forEach(element => {
+  let options = document.createElement('option');
+  options.innerHTML = `S${element.season.toString().padStart(2,'0')}E${element.number.toString().padStart(2,'0')}-${element.name}`;
```

Comment:
One more thing: rather than setting the `innerHTML` for each option, it's better to set the `option.text` (that will be displayed to the user) and the `option.value` (that will be available in the change listener as your w.target.value).